eth/v1alpha1/node.proto

```diff
@@ -62,47 +47,54 @@ service Node {
             get: "/eth/v1alpha1/node/services"
         };
     }
-
-    // Retrieve the list of peers currently connected to this node.
-    rpc ListPeers(google.protobuf.Empty) returns (Peers) {
-        option (google.api.http) = {
-            get: "/eth/v1alpha1/node/peers"
-        };
-    }
-}
-
-// Information about the current network sync status of the node.
-message SyncStatus {
-    // Whether or not the node is currently syncing.
-    bool syncing = 1;
 }
 
-// Information about the genesis of Ethereum 2.0.
-message Genesis {
-    // UTC time specified in the chain start event in the deposit contract.
-    google.protobuf.Timestamp genesis_time = 1;
-
-    // Address of the deposit contract in the Ethereum 1 chain.
-    bytes deposit_contract_address = 2;
+message NodeInfo {
+    // A unique identifier for the node.
+    string node_id = 1;
+    // The version of the currently-running node software.
+    string version = 2;
+    // State of the node's interaction with its peers.
+    SyncState sync = 3;
+    // The address(es) where the node is listening for incoming libp2p connections.
+    repeated string addresses = 4;
+    // Information about nodes to which this peer is connected.
+    repeated Peer peers = 5;
```

Comment:
if this is a list of all peers, maybe it's worth replacing with a peer count and having the list of peers on a separate rpc call (like currently) so it keeps the response lightweight

Reply:
Computationally speaking, the main effort is in fetching the list of peers so sending per-peer info isn't much more expensive than sending just the number. It will increase the size of the output, but not to any massive extent.  I'd be inclined to keep it here to reduce the number of calls we have.

Reply:
unless you have a few hundred peers? over the wire time should increase then i guess